cmd/kube-apiserver/app/server.go

```diff
@@ -129,6 +134,9 @@ func Run(s *options.ServerRunOptions) error {
 	if err := s.Features.ApplyTo(genericConfig); err != nil {
 		return err
 	}
+	if err := utilwait.PollImmediate(etcdRetryInterval, etcdRetryLimit*etcdRetryInterval, preflight.EtcdConnection{ServerList: s.Etcd.StorageConfig.ServerList}.CheckEtcdServers); err != nil {
```

Comment:
How about putting this in GenericApiServer? Maybe in GenericApiServer.PrepareRun or Run?
This doesnt see specific to kube-apiserver. federation-apiserver, for ex, should also have this?